AEPMessaging/Sources/BatchedPropositionInteraction.swift

```diff
@@ -0,0 +1,102 @@
+/*
+ Copyright 2025 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import AEPServices
+import Foundation
+
+/// A struct that helps to build data for batched proposition interaction event
+struct BatchedPropositionInteraction {
+    /// Edge event type represented by enum `MessagingEdgeEventType`
+    let eventType: MessagingEdgeEventType
+
+    /// Interaction string to identify interaction type with the proposition items
+    let interaction: String?
+
+    /// Collection of proposition items
+    let propositionItems: [PropositionItem]
+
+    /// Generates the XDM for the batched proposition interaction
+    /// - Returns: XDM dictionary
+    func generateXDM() -> [String: Any] {
+        let propositionInteractionDict = createPropositionInteractions()
+
+        // Return empty dictionary if no valid propositions
+        guard !propositionInteractionDict.isEmpty else {
+            return [:]
+        }
+
+        var decisioning: [String: Any] = [
+            MessagingConstants.XDM.Inbound.Key.PROPOSITION_EVENT_TYPE: [
+                eventType.propositionEventType: 1
+            ],
+            MessagingConstants.XDM.Inbound.Key.PROPOSITIONS: propositionInteractionDict
+        ]
+
+        if let interaction = interaction {
+            var propActionDict: [String: String]?
+
+            switch eventType {
+            case .interact:
+                propActionDict = [
+                    MessagingConstants.XDM.Inbound.Key.ID: interaction,
+                    MessagingConstants.XDM.Inbound.Key.LABEL: interaction
+                ]
+            case .suppressDisplay:
+                propActionDict = [
+                    MessagingConstants.XDM.Inbound.Key.REASON: interaction
```

Comment:
nit: We're only batching trigger events right? Wondering if this block will ever get executed

Reply:
We will also begin batching display events soon. 
However, I think it;s preferable to keep the BatchedPropositionInteraction class independent of what it’s batching and it’s best to manage all scenarios at this level.